src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventLogger.java

```diff
@@ -101,6 +103,7 @@ protected EventLogger(HiveConf conf, Clock clock) {
     recordsWriterFactory = createRecordsWriterFactory(baseDir, conf, clock);
     if (!isHiveVersionSupported(getHiveVersion()) || recordsWriterFactory == null) {
```

Comment:
Can we make a function such as -
```
private boolean shouldLog() {
  return isHiveVersionSupported(getHiveVersion()) && recordsWriterFactory != null;
}
```
rather than store it in `loggingEnabled` and use it inside the `handle` method?

Reply:
I moved out hive version check to the outer class completely, as without that hook wails during class instantiation on Hive 2.1.1 because of missing methods.